src/effects/VST3/VST3Effect.cpp

```diff
@@ -0,0 +1,549 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file VST3Effect.cpp
+
+  @author Vitaly Sverchinsky
+
+  @brief Part of Audacity VST3 module
+
+**********************************************************************/
+
+#include "VST3Effect.h"
+
+#include <stdexcept>
+#include <wx/log.h>
+#include <pluginterfaces/vst/ivstaudioprocessor.h>
+#include <pluginterfaces/vst/ivsteditcontroller.h>
+
+#include "VST3EffectsModule.h"
```

Comment:
There is an include dependency cycle between VST3Effect.h and VST3EffectsModule.h.  A cycle of just two files doesn't concern me much, but it would be nice to figure out the extra effort to break the cycle.  I think the module depends one-way on the effect, becuase it must be a factory of effects.

Reply:
The only reason why VST3EffectsModule is needed here is that VST3Effect duplicates itself when the effect requests to add a new realtime processor, no details about VST3EffectModule itself actually are needed here, so to make everything clearer I'll extract VST host access function from VST3EffectsModule to some other place